File: frontend/src/types/screening.ts
/** Mirrors backend/app/schemas/screening.py. */

export type ScreeningStatus = "PENDING" | "COMPLETED" | "FAILED";

export type ScreeningDecision = "MATCH" | "NOT_MATCH";

export interface ScreeningRunResponse {
  id: string;
  application_id: string;
  /** null = the automatic run at application submission. */
  requested_by_user_id: string | null;
  status: ScreeningStatus;
  provider: string;
  model: string;
  overall_score: number | null;
  recommendation: string | null;
  summary: string | null;
  matching_skills: string[] | null;
  missing_skills: string[] | null;
  strengths: string[] | null;
  concerns: string[] | null;
  experience_assessment: string | null;
  education_assessment: string | null;
  decision: ScreeningDecision | null;
  matched_requirements: string[] | null;
  missing_requirements: string[] | null;
  error_message: string | null;
  created_at: string;
  completed_at: string | null;
}

File: backend/app/schemas/screening.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.screening import ScreeningDecision, ScreeningStatus


class ScreeningRunResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    application_id: uuid.UUID
    requested_by_user_id: uuid.UUID | None = None
    status: ScreeningStatus
    provider: str
    model: str
    overall_score: int | None
    recommendation: str | None
    summary: str | None
    matching_skills: list[str] | None
    missing_skills: list[str] | None
    strengths: list[str] | None
    concerns: list[str] | None
    experience_assessment: str | None
    education_assessment: str | None
    decision: ScreeningDecision | None = None
    matched_requirements: list[str] | None = None
    missing_requirements: list[str] | None = None
    error_message: str | None
    created_at: datetime
    completed_at: datetime | None
